sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/RegionNameToRegionIdMap.java

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.routing;
+
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+public class RegionNameToRegionIdMap {
```

Comment:
Can you remind me what will happen when a region is sued (and routed to) that is not part of this map?

Reply:
The fallback is to use a session token which is representative of progress of all regions.